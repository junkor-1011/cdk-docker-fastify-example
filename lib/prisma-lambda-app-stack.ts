import {
  // Aws,
  // Duration,
  Stack,
  StackProps,
  aws_apigateway as apigateway,
  aws_lambda as lambda,
} from 'aws-cdk-lib';
// import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
import { Construct } from 'constructs';

const environmentForPrisma = {
  DATABASE_URL: 'postgres://dummy:5443/mydb',
};
/*
const commandHooksForPrisma = {
  beforeInstall(inputDir: string, outputDir: string): string[] {
    return [``];
  },
  beforeBundling(inputDir: string, outputDir: string): string[] {
    return [``];
  },
  afterBundling(inputDir: string, outputDir: string): string[] {
    return [
      `cd ${inputDir}/packages/fastify-app && pnpm prisma generate`,
      `cp ${inputDir}/packages/fastify-app/node_modules/.pnpm/prisma@4.6.1/node_modules/prisma/libquery_engine-rhel-openssl-1.0.x.so.node ${outputDir}`,
      `cp ${inputDir}/packages/fastify-app/prisma/schema.prisma ${outputDir}`,
    ];
  },
};
*/

export class PrismaLambdaAppStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    /*
    const fastifyAppLambda = new NodejsFunction(this, 'FastifyAppLambda', {
      entry: 'packages/fastify-app/lambda.ts',
      handler: 'handler',
      runtime: lambda.Runtime.NODEJS_16_X,
      bundling: {
        sourceMap: true,
        minify: true,
        commandHooks: commandHooksForPrisma,
      },
      environment: {
        ...environmentForPrisma,
        STAGE: 'PRODUCTION',
      },
    });
    */
    const fastifyAppLambda = new lambda.DockerImageFunction(this, 'fastifyImageFunction', {
      code: lambda.DockerImageCode.fromImageAsset('packages/fastify-app'),
      environment: {
        ...environmentForPrisma,
        STAGE: 'PRODUCTION',
      },
    });

    const helloLambda = new lambda.DockerImageFunction(this, 'helloImageFunction', {
      code: lambda.DockerImageCode.fromImageAsset('packages/hello-world'),
    });

    const fastapiLambda = new lambda.DockerImageFunction(this, 'fastapiImageFunction', {
      code: lambda.DockerImageCode.fromImageAsset('assets/fastapi-app'),
    });

    const lambdaApi = new apigateway.LambdaRestApi(this, 'fastifyAppApi', {
      handler: fastifyAppLambda,
      proxy: false,
    });
    const fastapiIntegration = new apigateway.LambdaIntegration(fastapiLambda);
    lambdaApi.root.addMethod('GET', fastapiIntegration);
    const docsPath = lambdaApi.root.addResource('docs');
    docsPath.addMethod('GET', fastapiIntegration);
    const openapiPath = lambdaApi.root.addResource('openapi.json');
    openapiPath.addMethod('GET', fastapiIntegration);

    const userPath = lambdaApi.root.addResource('user');
    userPath.addMethod('GET');
    userPath.addMethod('POST');
    const userIdPath = userPath.addResource('{id}');
    userIdPath.addMethod('GET');
    userIdPath.addMethod('PUT');
    userIdPath.addMethod('PATCH');
    userIdPath.addMethod('DELETE');

    const helloPath = lambdaApi.root.addResource('hello');
    helloPath.addMethod('GET', new apigateway.LambdaIntegration(helloLambda));
  }
}
