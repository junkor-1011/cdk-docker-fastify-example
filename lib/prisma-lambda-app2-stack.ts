import {
  // Aws,
  // Duration,
  Stack,
  StackProps,
  aws_apigateway as apigateway,
  aws_lambda as lambda,
} from 'aws-cdk-lib';
// import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
import { Construct } from 'constructs';

export class PrismaLambdaApp2Stack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const helloLambda = new lambda.DockerImageFunction(this, 'helloImageFunction', {
      code: lambda.DockerImageCode.fromImageAsset('packages/hello-world'),
    });

    const fastapiLambda = new lambda.DockerImageFunction(this, 'fastapiImageFunction', {
      code: lambda.DockerImageCode.fromImageAsset('assets/fastapi-app'),
    });

    const lambdaApi = new apigateway.LambdaRestApi(this, 'helloApi', {
      handler: helloLambda,
      proxy: false,
    });
    const fastapiIntegration = new apigateway.LambdaIntegration(fastapiLambda);
    lambdaApi.root.addMethod('GET', fastapiIntegration);
    const docsPath = lambdaApi.root.addResource('docs');
    docsPath.addMethod('GET', fastapiIntegration);
    const openapiPath = lambdaApi.root.addResource('openapi.json');
    openapiPath.addMethod('GET', fastapiIntegration);

    const helloPath = lambdaApi.root.addResource('hello');
    helloPath.addMethod('GET', new apigateway.LambdaIntegration(helloLambda));
    const helloPath2 = lambdaApi.root.addResource('hello2');
    helloPath2.addMethod('GET');
  }
}
