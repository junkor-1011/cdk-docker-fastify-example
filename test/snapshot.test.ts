import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { PrismaLambdaAppStack } from '../lib/prisma-lambda-app-stack';

test('snapshot test', () => {
  const app = new cdk.App();
  const stack = new PrismaLambdaAppStack(app, 'TestStack');
  const template = Template.fromStack(stack).toJSON();

  expect(template).toMatchSnapshot();
});
