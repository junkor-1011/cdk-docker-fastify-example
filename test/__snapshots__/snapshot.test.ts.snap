// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
{
  "Outputs": {
    "fastifyAppApiEndpointC51352AA": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "fastifyAppApi4B7A160F",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "fastifyAppApi4B7A160F": {
      "Properties": {
        "Name": "fastifyAppApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "fastifyAppApiAccountD8BAAB5C": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "fastifyAppApi4B7A160F",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "fastifyAppApiCloudWatchRole8AB1E821",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "fastifyAppApiCloudWatchRole8AB1E821": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "fastifyAppApiDeployment35BD0E1Ed559b1d581627fcd1d3ef8f15b29cad8": {
      "DependsOn": [
        "fastifyAppApiGET26ABCC2C",
        "fastifyAppApiuseridDELETEC4DEADF2",
        "fastifyAppApiuseridGET1F4C0B21",
        "fastifyAppApiuseridPATCHE922D388",
        "fastifyAppApiuseridPUT4A32714F",
        "fastifyAppApiuserid3A88A7E8",
        "fastifyAppApiuserGET75E9C1AB",
        "fastifyAppApiuserPOST8AE508AB",
        "fastifyAppApiuser93025BEC",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "fastifyAppApiDeploymentStageprod9CC17FB8": {
      "DependsOn": [
        "fastifyAppApiAccountD8BAAB5C",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "fastifyAppApiDeployment35BD0E1Ed559b1d581627fcd1d3ef8f15b29cad8",
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "fastifyAppApiGET26ABCC2C": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "helloImageFunction7B41B444",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "fastifyAppApi4B7A160F",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "fastifyAppApiGETApiPermissionTestStackfastifyAppApi1F207F31GETAB6E8064": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "helloImageFunction7B41B444",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/",
              {
                "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
              },
              "/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiGETApiPermissionTestTestStackfastifyAppApi1F207F31GET316291E8": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "helloImageFunction7B41B444",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/test-invoke-stage/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuser93025BEC": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "fastifyAppApi4B7A160F",
            "RootResourceId",
          ],
        },
        "PathPart": "user",
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "fastifyAppApiuserGET75E9C1AB": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "fastifyImageFunction7241F15F",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "fastifyAppApiuser93025BEC",
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "fastifyAppApiuserGETApiPermissionTestStackfastifyAppApi1F207F31GETuser1E5DEDDC": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/",
              {
                "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
              },
              "/GET/user",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuserGETApiPermissionTestTestStackfastifyAppApi1F207F31GETuser8104AF56": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/test-invoke-stage/GET/user",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuserPOST8AE508AB": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "fastifyImageFunction7241F15F",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "fastifyAppApiuser93025BEC",
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "fastifyAppApiuserPOSTApiPermissionTestStackfastifyAppApi1F207F31POSTuserE705F496": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/",
              {
                "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
              },
              "/POST/user",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuserPOSTApiPermissionTestTestStackfastifyAppApi1F207F31POSTuserAA9300FE": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/test-invoke-stage/POST/user",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuserid3A88A7E8": {
      "Properties": {
        "ParentId": {
          "Ref": "fastifyAppApiuser93025BEC",
        },
        "PathPart": "{id}",
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "fastifyAppApiuseridDELETEApiPermissionTestStackfastifyAppApi1F207F31DELETEuserid1D2B47ED": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/",
              {
                "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
              },
              "/DELETE/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuseridDELETEApiPermissionTestTestStackfastifyAppApi1F207F31DELETEuseridE0606C52": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/test-invoke-stage/DELETE/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuseridDELETEC4DEADF2": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "DELETE",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "fastifyImageFunction7241F15F",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "fastifyAppApiuserid3A88A7E8",
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "fastifyAppApiuseridGET1F4C0B21": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "fastifyImageFunction7241F15F",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "fastifyAppApiuserid3A88A7E8",
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "fastifyAppApiuseridGETApiPermissionTestStackfastifyAppApi1F207F31GETuserid38DC0690": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/",
              {
                "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
              },
              "/GET/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuseridGETApiPermissionTestTestStackfastifyAppApi1F207F31GETuserid1626AC8F": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/test-invoke-stage/GET/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuseridPATCHApiPermissionTestStackfastifyAppApi1F207F31PATCHuserid147BE162": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/",
              {
                "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
              },
              "/PATCH/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuseridPATCHApiPermissionTestTestStackfastifyAppApi1F207F31PATCHuserid418A0B5E": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/test-invoke-stage/PATCH/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuseridPATCHE922D388": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "PATCH",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "fastifyImageFunction7241F15F",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "fastifyAppApiuserid3A88A7E8",
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "fastifyAppApiuseridPUT4A32714F": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "PUT",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "fastifyImageFunction7241F15F",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "fastifyAppApiuserid3A88A7E8",
        },
        "RestApiId": {
          "Ref": "fastifyAppApi4B7A160F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "fastifyAppApiuseridPUTApiPermissionTestStackfastifyAppApi1F207F31PUTuserid090CB605": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/",
              {
                "Ref": "fastifyAppApiDeploymentStageprod9CC17FB8",
              },
              "/PUT/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyAppApiuseridPUTApiPermissionTestTestStackfastifyAppApi1F207F31PUTuseridD126A85D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "fastifyImageFunction7241F15F",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "fastifyAppApi4B7A160F",
              },
              "/test-invoke-stage/PUT/user/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "fastifyImageFunction7241F15F": {
      "DependsOn": [
        "fastifyImageFunctionServiceRole02502A60",
      ],
      "Properties": {
        "Code": {
          "ImageUri": {
            "Fn::Sub": "\${AWS::AccountId}.dkr.ecr.\${AWS::Region}.\${AWS::URLSuffix}/cdk-hnb659fds-container-assets-\${AWS::AccountId}-\${AWS::Region}:845f09d78ace0546740ab9c03c3ed7e4f2a38339b8e85941a0c3c552f5665d82",
          },
        },
        "Environment": {
          "Variables": {
            "DATABASE_URL": "postgres://dummy:5443/mydb",
            "STAGE": "PRODUCTION",
          },
        },
        "PackageType": "Image",
        "Role": {
          "Fn::GetAtt": [
            "fastifyImageFunctionServiceRole02502A60",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "fastifyImageFunctionServiceRole02502A60": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "helloImageFunction7B41B444": {
      "DependsOn": [
        "helloImageFunctionServiceRoleFA0AE26A",
      ],
      "Properties": {
        "Code": {
          "ImageUri": {
            "Fn::Sub": "\${AWS::AccountId}.dkr.ecr.\${AWS::Region}.\${AWS::URLSuffix}/cdk-hnb659fds-container-assets-\${AWS::AccountId}-\${AWS::Region}:e5948586eeb3c45e2f79e0634039a5046d9c6de8201822426a18bf72ed43063d",
          },
        },
        "PackageType": "Image",
        "Role": {
          "Fn::GetAtt": [
            "helloImageFunctionServiceRoleFA0AE26A",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "helloImageFunctionServiceRoleFA0AE26A": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
